rust/src/enhancers/actions.rs

```diff
@@ -46,46 +46,99 @@ pub struct FlagAction {
 }
 
 impl FlagAction {
-    fn iter_frames<'f>(
+    fn slice_to_range<'f, I>(&self, items: &'f [I], idx: usize) -> impl Iterator<Item = &'f I> {
+        let slice = match self.range {
+            Some(Range::Up) => items.get(idx + 1..),
+            Some(Range::Down) => items.get(..idx),
+            None => items.get(idx..idx + 1),
+        };
+        slice.unwrap_or_default().iter()
+    }
+
+    fn slice_to_range_mut<'f, I>(
         &self,
-        frames: &'f mut [Frame],
+        items: &'f mut [I],
         idx: usize,
-    ) -> impl Iterator<Item = &'f mut Frame> {
+    ) -> impl Iterator<Item = &'f mut I> {
         let slice = match self.range {
-            Some(Range::Up) => frames.get_mut(idx + 1..),
-            Some(Range::Down) => frames.get_mut(..idx),
-            None => frames.get_mut(idx..idx + 1),
+            Some(Range::Up) => items.get_mut(idx + 1..),
+            Some(Range::Down) => items.get_mut(..idx),
+            None => items.get_mut(idx..idx + 1),
         };
         slice.unwrap_or_default().iter_mut()
     }
-}
 
-impl fmt::Display for FlagAction {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if let Some(range) = self.range {
-            write!(f, "{range}")?;
+    /// Applies this action's modification to the given list of frames at the given index.
+    pub fn apply_modifications_to_frame(&self, frames: &mut [Frame], idx: usize) {
+        if self.ty == FlagActionType::App {
+            for frame in self.slice_to_range_mut(frames, idx) {
+                let orig_in_app = frame.in_app;
+                if orig_in_app != self.flag {
+                    frame.in_app = self.flag;
+                    frame.orig_in_app = Some(orig_in_app);
+                }
```

Comment:
The problem here is rather that is `create_match_frame`, we default `in_app` to false, so we do not cover the case it is `None`.